erigon_vm crashes on statetest execution

#### System information

Erigon version: `./erigon --version`

```
erigon_vm version 3.01.0-dev-2a81add7
```

Commit hash: `2a81add7`


### Description

It crashes on execution of statetests (with json logging). As far as I can tell, it crashes on any/all tests, so should be fairly easy to repro. (the particular testcase below comes from https://github.com/holiman/goevmlab/tree/master/evms/testdata/cases) 
```
docker run -it -v /home/user/workspace/goevmlab/evms/testdata/:/testdata  holiman/omnifuzz

root@fb4d809553b1:/# /erigon_vm --json --noreturndata --nomemory statetest /testdata/cases/00000936-mixed-1.json
{"output":"","gasUsed":"0x79bc70"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x18440e3]

goroutine 1 [running]:
github.com/erigontech/erigon/eth/tracers/logger.(*JSONLogger).OnOpcode(0xc001772900, 0x0, 0x60, 0x79bc70, 0x3, {0x23b03d8, 0xc000de6888}, {0x0, 0x0, 0x0}, ...)
	github.com/erigontech/erigon/eth/tracers/logger/logger_json.go:81 +0xa3
github.com/erigontech/erigon/core/vm.(*EVMInterpreter).Run(0xc000de6858, 0xc0012d60d0, {0xc001651ac0, 0x3a, 0x3a}, 0x0)
	github.com/erigontech/erigon/core/vm/interpreter.go:365 +0x96d
github.com/erigontech/erigon/core/vm.run(...)
	github.com/erigontech/erigon/core/vm/evm.go:65
github.com/erigontech/erigon/core/vm.(*EVM).call(0xc001872580, 0xf1, {0x238ef20, 0xc001794b70}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	github.com/erigontech/erigon/core/vm/evm.go:281 +0x1516
github.com/erigontech/erigon/core/vm.(*EVM).Call(...)
	github.com/erigontech/erigon/core/vm/evm.go:307
github.com/erigontech/erigon/core.(*StateTransition).TransitionDb(0xc0007b8480, 0x1, 0x0)
	github.com/erigontech/erigon/core/state_transition.go:688 +0x1bf4
github.com/erigontech/erigon/core.ApplyMessage(0xc001872580, {0x23bc220, 0xc000786280}, 0xc0012a81d0, 0x1, 0x0, {0x0, 0x0})
	github.com/erigontech/erigon/core/state_transition.go:151 +0x265
github.com/erigontech/erigon/tests.(*StateTest).RunNoVerify(_, {_, _}, {{_, _}, _}, {0xc0008e9cc0, 0x0, 0x0, 0x0, ...}, ...)
	github.com/erigontech/erigon/tests/state_test_util.go:272 +0x127c
github.com/erigontech/erigon/tests.(*StateTest).Run(_, {_, _}, {{_, _}, _}, {0xc0008e9cc0, 0x0, 0x0, 0x0, ...}, ...)
	github.com/erigontech/erigon/tests/state_test_util.go:170 +0xbe
main.aggregateResultsFromStateTests(0xc0008ec690, {0xc0008e9cc0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	github.com/erigontech/erigon/cmd/evm/staterunner.go:167 +0xa88
main.runStateTest({0x7ffe27426cf8?, 0x0?}, {0xc0008e9cc0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	github.com/erigontech/erigon/cmd/evm/staterunner.go:119 +0xae
main.stateTestCmd(0xc0005fd880)
	github.com/erigontech/erigon/cmd/evm/staterunner.go:91 +0x946
github.com/urfave/cli/v2.(*Command).Run(0x33fc5a0, 0xc0005fd880, {0xc00178e220, 0x2, 0x2})
	github.com/urfave/cli/v2@v2.27.5/command.go:276 +0x7be
github.com/urfave/cli/v2.(*Command).Run(0xc0001cfce0, 0xc0006fcd80, {0xc00003e240, 0x6, 0x6})
	github.com/urfave/cli/v2@v2.27.5/command.go:269 +0xa45
github.com/urfave/cli/v2.(*App).RunContext(0xc000832600, {0x23a48a0, 0x36e4a80}, {0xc00003e240, 0x6, 0x6})
	github.com/urfave/cli/v2@v2.27.5/app.go:333 +0x5a5
github.com/urfave/cli/v2.(*App).Run(...)
	github.com/urfave/cli/v2@v2.27.5/app.go:307
main.main()
	github.com/erigontech/erigon/cmd/evm/main.go:197 +0x45
```
### version
```
root@fb4d809553b1:/# /erigon_vm --version
erigon_vm version 3.01.0-dev-2a81add7
```

--- a/tests/state_test_util.go
+++ b/tests/state_test_util.go
@@ -264,13 +264,20 @@ func (t *StateTest) RunNoVerify(tx kv.RwTx, subtest StateSubtest, vmconfig vm.Co
 		}
 	}
 	evm := vm.NewEVM(context, txContext, statedb, config, vmconfig)
+	if vmconfig.Tracer != nil && vmconfig.Tracer.OnTxStart != nil {
+		vmconfig.Tracer.OnTxStart(evm.GetVMContext(), nil, libcommon.Address{})
+	}
 
 	// Execute the message.
 	snapshot := statedb.Snapshot()
 	gaspool := new(core.GasPool)
 	gaspool.AddGas(block.GasLimit()).AddBlobGas(config.GetMaxBlobGasPerBlock(header.Time))
-	if _, err = core.ApplyMessage(evm, msg, gaspool, true /* refunds */, false /* gasBailout */, nil /* engine */); err != nil {
+	res, err := core.ApplyMessage(evm, msg, gaspool, true /* refunds */, false /* gasBailout */, nil /* engine */)
+	if err != nil {
 		statedb.RevertToSnapshot(snapshot)
+	}
+	if vmconfig.Tracer != nil && vmconfig.Tracer.OnTxEnd != nil {
+		vmconfig.Tracer.OnTxEnd(&types.Receipt{GasUsed: res.UsedGas}, nil)
 	}
 
 	if err = statedb.FinalizeTx(evm.ChainRules(), w); err != nil {

